src/error.rs

```diff
@@ -218,3 +218,91 @@ impl From<iota_ledger::api::errors::APIError> for Error {
         }
     }
 }
+
+impl serde::Serialize for Error {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        fn serialize_variant<S: Serializer>(
+            error: &Error,
+            serializer: S,
+            variant_name: &str,
+        ) -> std::result::Result<S::Ok, S::Error> {
+            let mut state = serializer.serialize_struct("Error", 2)?;
+            state.serialize_field("type", variant_name)?;
+            state.serialize_field("error", &error.to_string())?;
+            state.end()
+        }
+
+        match self {
```

Comment:
Instead of mapping all manually, is there no macro that can do that for us?

Reply:
Good point, I was going to use `#[serde(tag = "type", content = "error")` but I ran into errors because some of the variants don't implement serialize, `FromHexError(#[from] hex::FromHexError),`, `CommonError(#[from] bee_common::logger::Error)` and a few others.

https://github.com/iotaledger/iota.rs/blob/dev/src/error.rs#L56

Reply:
But for this case we don't care about the errors inside anyways, so maybe we can use the enum discriminant with this approach https://stackoverflow.com/a/32712140 or something similar?

Reply:
I think I see what you're saying, we don't need the errors inside but the error message generated by `thiserror`, so maybe the manual `Serialize` implementation don't have to be exhaustive, right?